src/frontend/src/components/SearchPage/SearchResults.jsx

```diff
@@ -23,15 +23,18 @@ const useStyles = makeStyles(() => ({
 const SearchResults = ({ text, filter }) => {
   const classes = useStyles();
   const { telescopeUrl } = useSiteMetadata();
-  const url = `${telescopeUrl}/query?text=${encodeURIComponent(text)}&filter=${filter}`;
+  const url = `${telescopeUrl}/query?text=${encodeURIComponent(text)}&filter=${filter}&page=`;
 
   // We only bother doing the request if we have something to search for.
   const shouldFetch = () => text.length > 0;
-  const { data, loading, error } = useSWR(shouldFetch() ? url : null, async (u) => {
-    const res = await fetch(u);
-    const results = await res.json();
-    return results.values;
-  });
+  const { data, size, setSize, loading, error } = useSWRInfinite(
+    (index) => (shouldFetch() ? url + index : null),
```

Comment:
Your URL will need to include size info, so we ask for the right offset.  We can't send `&page=` (i.e., with no value) all the time.

Reply:
by size info/offset do you mean the number of results per fetch? also im under the impression that page= will have a value because of the index argument in the getKey function for swr infinite. sorry I'm still learning how to use the swr react hook so i might be a little confused, what would you suggest I change here?

Reply:
I see, you're actually doing it correctly, but it's not very easy to read.  Let's do this via a function call:

```js
const prepareUrl = (index) => ${telescopeUrl}/query?text=${encodeURIComponent(text)}&filter=${filter}&page=${index + 1}`;
...
const { data, size, setSize, loading, error } = useSWRInfinite(
    (index) => (shouldFetch() ? prepareUrl(index) : null),
```